### PR TITLE
[Feature] support create iceberg table

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
Support create iceberg table under iceberg catalog. This table will sync to the user meta service. The Spark/hive can also query it.
usage:
- `create external table iceberg_catalog.iceberg_db.iceberg_table (c1 int, c2 int);`
- `create external table iceberg_db.iceberg_table (c1 int, c2 int);  // current context catalog is iceberg catalog`
- `create external table iceberg_table (c1 int, c2 int); // current context catalog and db are both iceberg.`
- `create external table iceberg_table (c1 int, c2 int ) engine = iceberg;`
- `create external table iceberg_table (c1 int, c2 int, c3 int ) engine = iceberg partition by (c2, c3);`
- `create external table iceberg_table (c1 int, c2 int ) partition by (c2) properties ("location"="hdfs://hadoop:9000/user/warehouse/iceberg_db.db/iceberg_table", "file_format"="parquet");`

https://github.com/StarRocks/starrocks/issues/21502

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergApiConverter.java [hunk 1]
```diff
@@ -63,6 +71,64 @@ public static IcebergTable toIcebergTable(Table nativeTbl, String catalogName, S
...
+        for (Column column : columns) {
+            int index = icebergColumns.size();
+            org.apache.iceberg.types.Type type = toIcebergColumnType(column.getType());
+            Types.NestedField field = Types.NestedField.of(
+                    index, column.isAllowNull(), column.getName(), type, column.getComment());
+            icebergColumns.add(field);
+        }
+
+        org.apache.iceberg.types.Type icebergSchema = Types.StructType.of(icebergColumns);
+        AtomicInteger nextFieldId = new AtomicInteger(1);
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergApiConverter.java:85]: Why is atomic type needed?

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergCatalog.java [hunk 2]
```diff
@@ -29,54 +32,45 @@
...
+    List<TableIdentifier> listTables(Namespace of);
+
+    default Transaction newCreateTableTransaction(
+            String dbName,
+            String tableName,
+            Schema schema,
+            PartitionSpec partitionSpec,
+            String location,
+            Map<String, String> properties) {
+        return null;
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergCatalog.java:69]: why not throw exception

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergCatalog.java]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 3]
```diff
@@ -1966,6 +1966,7 @@ optimizerTrace
 partitionDesc
     : PARTITION BY RANGE identifierList '(' (rangePartitionDesc (',' rangePartitionDesc)*)? ')'
     | PARTITION BY LIST identifierList '(' (listPartitionDesc (',' listPartitionDesc)*)? ')'
+    | PARTITION BY identifierList
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1969]: this grammar can cover hive/hudi/iceberg/deltalake creating table

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergApiConverter.java [hunk 4]
```diff
@@ -127,4 +193,24 @@ private static TIcebergSchemaField getTIcebergSchemaField(Types.NestedField nest
         }
         return tIcebergSchemaField;
     }
+
+    public static Map<String, String> rebuildCreateTableProperties(Map<String, String> createProperties) {
+        ImmutableMap.Builder<String, String> tableProperties = ImmutableMap.builder();
+        createProperties.entrySet().stream()
+                .filter(entry -> createProperties.containsKey(entry.getKey()))
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergApiConverter.java:200]: This always true

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergApiConverter.java [hunk 5]
```diff
@@ -63,6 +71,64 @@ public static IcebergTable toIcebergTable(Table nativeTbl, String catalogName, S
...
+    }
+
+    public static org.apache.iceberg.types.Type toIcebergColumnType(Type type) {
+        PrimitiveType primitiveType = type.getPrimitiveType();
+
+        switch (primitiveType) {
+            case BOOLEAN:
+                return Types.BooleanType.get();
+            case INT:
+            case BIGINT:
```
[the remark this answers is not in the document]
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergApiConverter.java]: done

**Review by Astralidea — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**